### PR TITLE
Merge Master with 8.x

### PR DESCRIPTION
**Pull request by Sarangan0219**

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 name: Test Coverage Uploader
 
 on:
-  push:
-    branches:
-      - master
+  schedule:
+    - cron:  '0 20 * * *'
```
Comment (malinthaprasan) [inline .github/workflows/coverage.yml:5]: Lets stop running this workflow:
```
# added a workaround to stop running this
on:
  push:
    paths:
    - '**.never.ever.run'
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,32 +1,107 @@
 name: API Manager Build with Test
 
-on: [push, pull_request]
+on:
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+    branches:
+      - master
```
Comment (malinthaprasan) [inline .github/workflows/main.yml:9]: Same here.

Bdw we can add a new github workflow to build PRs and run with tests.

**Review by malinthaprasan — APPROVED** [review]

**Review by pubudu538 — APPROVED** [review]

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (Sarangan0219) [reply .github/workflows/coverage.yml]: Fixed https://github.com/wso2/carbon-apimgt/pull/10126/commits/f31a26dbfdd8436f0c2a6d20b334441d2772a183

File: .github/workflows/main.yml (hunk 2 above)
Comment (Sarangan0219) [reply .github/workflows/main.yml]: https://github.com/wso2/carbon-apimgt/pull/10126/commits/f31a26dbfdd8436f0c2a6d20b334441d2772a183

**Pull request merged**